### PR TITLE
✨ Update authentication scheme

### LINKED ISSUE #49: 2nd Factor Auth. - Google Authenticator
First off - thanks for this awesome project - I think it will be really useful for lots of folks.

I am having issues getting this to work with 2-factor authentication scheme on an HPC system. Our system uses Google Authenticator for a 6-digit code. The SSH login process looks like:

```bash
ssh username@host
Verification code:
Password:
```
When I try jupyter-forward I like:
```
jupyter-forward rowan.gaffney@ceres.scinet.usda.gov --launch-command "sbatch XXXXXXX" --notebook-dir=/lustre/project/cper_neon_aop/ --port=8889
```

<details>
  <summary>I get the this error</summary>

```
Password:
Traceback (most recent call last):
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/paramiko/transport.py", line 1528, in auth_password
    return self.auth_interactive(username, handler)
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/paramiko/transport.py", line 1633, in auth_interactive
    return self.auth_handler.wait_for_response(my_event)
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/paramiko/auth_handler.py", line 250, in wait_for_response
    raise e
paramiko.ssh_exception.AuthenticationException: Authentication failed.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/rowangaffney/miniconda3/bin/jupyter-forward", line 10, in <module>
    sys.exit(main())
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/jupyter_forward/cli.py", line 97, in main
    typer.run(app())
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/typer/main.py", line 214, in __call__
    return get_command(self)(*args, **kwargs)
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/typer/main.py", line 497, in wrapper
    return callback(**use_params)  # type: ignore
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/jupyter_forward/cli.py", line 91, in start
    shell=shell,
  File "<string>", line 11, in __init__
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/jupyter_forward/core.py", line 52, in __post_init__
    self.session.open()
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/fabric/connection.py", line 634, in open
    self.client.connect(**kwargs)
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/paramiko/client.py", line 446, in connect
    passphrase,
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/paramiko/client.py", line 764, in _auth
    raise saved_exception
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/paramiko/client.py", line 751, in _auth
    self._transport.auth_password(username, password)
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/paramiko/transport.py", line 1531, in auth_password
    raise e
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/paramiko/transport.py", line 1509, in auth_password
    return self.auth_handler.wait_for_response(my_event)
  File "/home/rowangaffney/miniconda3/lib/python3.7/site-packages/paramiko/auth_handler.py", line 250, in wait_for_response
    raise e
paramiko.ssh_exception.BadAuthenticationType: Bad authentication type; allowed types: ['keyboard-interactive']
```
</details>

Seems like an authentication issue. I tried to work thru fabric --> paramiko authentication workflow, but had no luck. Then I saw this #7, but the solution was posted in that issue.

Tagging @andersy005 b/c I think you might be able to help me with this - thanks!

### PR DESCRIPTION
**Pull request by mnlevy1981**
Rather than rely on `connect_kwargs['password']` to send password to `Connection`, use a lightweight handler that is basically a lightweight version of the example in https://stackoverflow.com/a/43943658. This has the added bonus of the password prompt reflecting what you get from `ssh`... e.g. Casper requests `Token_Response:` and if you have your ssh keys configured to avoid requiring a password then you should no longer get any prompt at all (rather than a prompt that is not actually passed to the host).

Fixes #49

### CODE REVIEW

**Comment by mnlevy1981** [comment]
I'd vote for a squash-merge on this one as an API change I made in 85a05b5 was undone in 2134078 and we don't need to carry those poor decisions around :)

**Review by andersy005 — APPROVED** [review]
This looks great 👍... I'm going to wait for @rmg55's confirmation that this is working properly before merging...

**Comment by rmg55** [comment]
This solved the authentication issue I raised in #49 . Thanks of bunch @andersy005 and @mnlevy1981

**Comment by andersy005** [comment]
Many thanks @mnlevy1981 & @rmg55! Merging this....

**Pull request merged**
